src/skills-builder/skills-builder-modal/select-preferences/JobTitleAutosuggest.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Form,
+} from '@edx/paragon';
+import { logError } from '@edx/frontend-platform/logging';
+import { useAlgoliaSearch } from '../../utils/search';
+
+const JobTitleAutosuggest = ({ onSelected }) => {
+  // We only need to destructure the job search index here
+  const [,, jobSearchIndex] = useAlgoliaSearch();
+
+  const [jobInput, setJobInput] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [jobList, setJobList] = useState([]);
+
+  const queryForJobs = (query = '') => {
+    jobSearchIndex.search(query).then(({ hits }) => {
+      setJobList(hits);
+      setIsLoading(false);
+    }).catch(err => {
+      logError(err);
+      setJobList([]);
+      setIsLoading(false);
+    });
+  };
+
+  const handleAutoSuggestChange = (title) => {
+    setIsLoading(true);
+    setJobInput(title);
+    queryForJobs(title);
+  };
+
+  useEffect(() => {
+    queryForJobs();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <Form.Autosuggest
+      isLoading={isLoading}
+      value={jobInput}
+      onSelected={onSelected}
+      onChange={handleAutoSuggestChange}
+      name="job-title-suggest"
+    >
+      {jobList.map(job => (
+        <Form.AutosuggestOption key={job.id}>
+          {job.name}
+        </Form.AutosuggestOption>
+      ))}
+    </Form.Autosuggest>
+  );
+};
+
+JobTitleAutosuggest.propTypes = {
+  onSelected: PropTypes.func.isRequired,
+};
+
+export default JobTitleAutosuggest;
```

Comment:
Instead of making our own component for this... would this be a good place to use the <InstantSearch /> package and component that B2B uses? 

We would already have access to the client/index we would pass into the object and then it's less code for us to maintain.

Reply:
Currently working on this and believe it may provide a solution to some of the other concerns as well 👍